insonmnia/miner/network/tinc_driver.go

```diff
@@ -69,9 +90,58 @@ type TincNetworkState struct {
 	config   *TincNetworkConfig
 	mu       sync.RWMutex
 	cli      *client.Client
-	networks map[string]*TincNetwork
-	pools    map[string]*net.IPNet
+	Networks map[string]*TincNetwork
+	Pools    map[string]*net.IPNet
 	logger   *zap.SugaredLogger
+	storage  store.Store
+}
+
+func (t *TincNetworkState) load() (err error) {
+	defer func() {
+		if err == store.ErrKeyNotFound {
+			err = nil
+		}
+		if err != nil {
+			t.logger.Errorf("could not load tinc network state - %s; erasing key", err)
+			delErr := t.storage.Delete("state")
+			if delErr != nil {
+				t.logger.Errorf("could not cleanup storage for tinc network - %s", delErr)
+			}
+		}
+	}()
+
+	exists, err := t.storage.Exists("state")
+	if err != nil || !exists {
+		return err
+	}
+
+	data, err := t.storage.Get("state")
+	if err != nil {
+		return err
+	}
+
+	err = json.Unmarshal(data.Value, t)
+	for _, n := range t.Networks {
+		n.cli = t.cli
```

Comment:
What will be in `t.cli` if unmarshalling is failed? Looks like you can alter the structure internals without error checking.

Reply:
In general that does not matter, but you are right - that would be useless